File: backend/app/models/reservation_attachment.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class ReservationAttachment(Base):
    __tablename__ = "reservation_attachments"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    reservation_id: Mapped[int] = mapped_column(Integer, ForeignKey("reservations.id", ondelete="CASCADE"), index=True)
    file_name: Mapped[str] = mapped_column(String(255), unique=True)
    original_name: Mapped[str] = mapped_column(String(255))
    content_type: Mapped[str] = mapped_column(String(100))
    size_bytes: Mapped[int] = mapped_column(Integer)
    category: Mapped[str] = mapped_column(String(30), default="other")
    uploaded_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

File: frontend/src/types/reservationAttachment.ts
export type AttachmentCategory = 'contract' | 'receipt' | 'photo' | 'media_consent' | 'other';

export interface ReservationAttachment {
  id: number;
  reservation_id: number;
  file_name: string;
  original_name: string;
  content_type: string;
  size_bytes: number;
  category: AttachmentCategory;
  url: string;
  uploaded_at: string;
}
